kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java

```diff
@@ -151,6 +152,11 @@ public String getActiveProcessImage(String containerId, long procInstId, String
                 populateSubProcessLink(containerId, activeNode, subProcessLinks);
             });
 
+            fullLogs.stream().filter(node -> ((org.jbpm.kie.services.impl.model.NodeInstanceDesc) node).getType() != 0 && ((org.jbpm.kie.services.impl.model.NodeInstanceDesc) node).getType() != 1)
```

Comment:
Plus, I think it type cast should not be needed since fullLogs is of type NodeInstanceDesc. At least you don't need to use fully qualified name, cause it's imported already.

Reply:
the type is int, not a String, the type means the state of the node. 
the class import "import org.jbpm.services.api.model.NodeInstanceDesc;", it's not org.jbpm.kie.services.impl.model.NodeInstanceDesc, and type just exited in new one.
we already get actvie and completed node from getProcessInstanceHistoryActive and getProcessInstanceHistoryCompleted, but abort, error we miss them. so I want get these kind of node.

Reply:
correct, returns int indeed, I mixed it with getNodeType.

Isn't there a set of constants for the int values? This way it's hard to decipher.

Reply:
https://github.com/kiegroup/jbpm/blob/afa412ae27f2b220c0db166c43badde51d529cec/jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/RuntimeDataService.java#L54
I think you are right, I will use this one.